x-pack/plugins/observability_solution/infra/common/http_api/custom_dashboards_api.ts

```diff
@@ -12,36 +12,48 @@ const AssetTypeRT = rt.type({
   assetType: ItemTypeRT,
 });
 
-const CustomDashboardRT = rt.intersection([
-  AssetTypeRT,
-  rt.type({
-    dashboardIdList: rt.array(rt.string),
-  }),
-  rt.partial({
-    kuery: rt.string,
-  }),
-]);
+const PayloadRT = rt.type({
+  dashboardSavedObjectId: rt.string,
+  dashboardFilterAssetIdEnabled: rt.boolean,
+});
+
+const existingIdRT = rt.partial({
```

Comment:
nit - other io-ts types are capitalized

```suggestion
const SavedObjectIdRT = rt.partial({
```

Reply:
I renamed it to `ExistingIdRT` because we have `SavedObjectIdRT` already